### PR TITLE
Animated

### PR DESCRIPTION
**Pull request by YuZhangYuanYYZ**
implement the favorate function.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +33,10 @@
   },
   "devDependencies": {
     "moment": "^2.24.0",
+    "react-addons-css-transition-group": "^15.6.2",
```
Comment (lanmengyujing) [inline package.json]: this library is not used

File: src/components/CommonTodo/View.js [hunk 2]
```diff
@@ -31,7 +30,7 @@ class View extends React.Component {
     >
     {
      <li style={{"width":this.props.showSideBar?"80%":"100%"}} key={this.props.todo.id} data-index={this.props.index} onDoubleClick={()=>this.sidbarTogle(this.props.todo.id)} className={this.convertDataToClassName(this.props.todo)}>{this.props.todo.text}
```
Comment (lanmengyujing) [inline src/components/CommonTodo/View.js]: this should be changed, I mentioned in other PR previously

File: src/components/RightSideBar/View.js [hunk 3]
```diff
@@ -34,31 +31,52 @@ class View extends React.Component {
...
-            <DatePicker selected = {
-                this.state.startDate
-            }   
-            onChange = {
-                this.onHandleChange
+            <CSSTransition
+            in={this.props.showSideBar}
+            timeout={300}
+            classNames="animated"
+            // unmountOnExit
```
Comment (lanmengyujing) [inline src/components/RightSideBar/View.js]: why this is comment out?

File: src/components/RightSideBar/style.css [hunk 4]
```diff
@@ -1,10 +1,29 @@
...
-}
\ No newline at end of file
+    
+    }
+    .animated-exit-done{
+        display:none;
+    }
+    /* enter是入场前的刹那（点击按钮），appear指页面第一次加载前的一刹那（自动） */
+    .animated-enter {
+    opacity: 0;
```
Comment (lanmengyujing) [inline src/components/RightSideBar/style.css]: indent is not correct

File: src/components/RightSideBar/style.css [hunk 5]
```diff
@@ -1,10 +1,29 @@
...
+    }
+    .animated-enter-active {
+    opacity: 1;
```
Comment (lanmengyujing) [inline src/components/RightSideBar/style.css]: indent is not correct

File: src/components/functionButtons/Container.js [hunk 6]
```diff
@@ -14,6 +14,9 @@ const mapDispatchToProps = dispatch => {
         onCompleteToggle: payload => {
             dispatch(completeToggle(payload))
         },
+        onFavorate: payload => {
```
Comment (lanmengyujing) [inline src/components/functionButtons/Container.js]: onFaviourate

File: src/components/functionButtons/View.js [hunk 7]
```diff
@@ -76,6 +106,15 @@ class View extends React.Component {
                     </svg>
 
                 </button>
+                <button className={this.props.todo.favorate?"favorateButtonClicked":"favorateButton"} onClick={this.favorateHandler} >
+                    <svg version="1.1" xmlns="http://www.w3.org/2000/svg" xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 22 22" style={{ enablebackground: "new 0 0 22 2" }} space="preserve">
```
Comment (lanmengyujing) [inline src/components/functionButtons/View.js]: svg should be moved to individual file

File: src/components/functionButtons/View.js [hunk 8]
```diff
@@ -53,6 +55,34 @@ class View extends React.Component {
...
+    favorateHandler(e) {
+        let grandParent = e.target.parentNode.parentNode;
+        let dataIndex = Number(grandParent.dataset.index);
+      
+        if (!isNaN(dataIndex)) {
+            
+            const todo = this.props.todos[dataIndex];
+            
+            this.props.onFavorate(todo);
+           // console.log( e.target.className,todo.favorate," e.target.className,todo.favorate")
```
Comment (lanmengyujing) [inline src/components/functionButtons/View.js]: please delete console.log

File: src/components/functionButtons/View.js [hunk 9]
```diff
@@ -53,6 +55,34 @@ class View extends React.Component {
...
+                method: 'PUT',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({
+                    ...todo,
+                    favorate: finalFavorate
+                })
+            }
+            this.changeTodosState(this.props.todos[dataIndex].id, putOptions);
```
Comment (lanmengyujing) [inline src/components/functionButtons/View.js]: this method is undefined

File: package.json (hunk 1 above)
Comment (YuZhangYuanYYZ) [reply package.json]: accept and changed

Comment (YuZhangYuanYYZ) [reply package.json]: i mean I delete it

File: src/components/RightSideBar/View.js (hunk 3 above)
Comment (YuZhangYuanYYZ) [reply src/components/RightSideBar/View.js]: delete

File: src/components/RightSideBar/style.css (hunk 4 above)
Comment (YuZhangYuanYYZ) [reply src/components/RightSideBar/style.css]: OK

File: src/components/functionButtons/Container.js (hunk 6 above)
Comment (YuZhangYuanYYZ) [reply src/components/functionButtons/Container.js]: what?

File: src/components/functionButtons/View.js (hunk 7 above)
Comment (YuZhangYuanYYZ) [reply src/components/functionButtons/View.js]: how to?

File: src/components/functionButtons/View.js (hunk 9 above)
Comment (YuZhangYuanYYZ) [reply src/components/functionButtons/View.js]: bullshit

**Pull request merged**